### PR TITLE
Week2 json

### PR DESCRIPTION
**Pull request by BenMob**
# Json
## This branch contains a demo of how to transport Data from the server to the client in a Json formatted string.

**Note:** For some reason, commits from the previous PR also appear here. I think i messed up somewhere. The commit named `Json Demo` is the one containing the changes described here.

**Screenshot** 
Screen Shot 2020-06-11 at 8 41 44 PM

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -15,18 +15,47 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import com.google.gson.Gson;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.List;
```
Comment (gitmnk) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:23]: List below ArrayList

File: portfolio/src/main/webapp/css/style.css [hunk 2]
```diff
@@ -20,7 +20,7 @@ a:hover{
 .content{
     color:whitesmoke;
     bottom: 0%;
-    width: 960px;
+    width: 80%;
```
Comment (gitmnk) [inline portfolio/src/main/webapp/css/style.css:23]: good job fixing this!

File: portfolio/src/main/webapp/js/script.js [hunk 3]
```diff
@@ -12,60 +12,56 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/********************************************************
- * Listens to the about link in the navigation bar and
- * turns the text to yellow when the about section is visited.  
+/*****************************************************************
+ * This function requests data from the server using async await
+ * @param url: The url to request from 
+ * @param containerId: the id of the conatiner in which this data is
```
Comment (gitmnk) [inline portfolio/src/main/webapp/js/script.js]: typo container

File: portfolio/src/main/webapp/js/script.js [hunk 4]
```diff
@@ -12,60 +12,56 @@
...
-    let projects = document.getElementById('projects-link')
+ async function fetchDataAsync(url, containerId){
+     const response = await fetch(url)
+     const data = await response.json()
+     presentData(data, containerId)
+     console.log(data)
+ }
+ 
+/****************************************************************
+ *This fcuntion processes reponses from the server appropriatelly
```
Comment (gitmnk) [inline portfolio/src/main/webapp/js/script.js]: typo fcuntion

Comment (gitmnk) [inline portfolio/src/main/webapp/js/script.js]: typo reponses

File: portfolio/src/main/webapp/js/script.js [hunk 5]
```diff
@@ -12,60 +12,56 @@
...
+ * and dpresents displays them on the page
+ * @param data: an array containing the data returned from the
+ *                 server 
+ *********/
+function presentData(data, containerId){
+    const container = document.getElementById(containerId)
+    let ul = document.createElement('ul');
+    addClass(ul, 'w3-ul', 'w3-round-16') // Adds some classes to the ul for styling
```
Comment (gitmnk) [inline portfolio/src/main/webapp/js/script.js:37]: nice use of styles

File: portfolio/src/main/webapp/js/script.js [hunk 6]
```diff
@@ -12,60 +12,56 @@
...
+     addClass(tag, 'w3-border-black') // Adds a class to the li for styling
+     return tag
+ }
 
+/***************************************************************
+ * This funtion adds a classes insite 
+ * @param htmlTag: Any HTML tag or element 
+ * @param ...classes: list of classes you wnat to add
+ *********/
+function addClass(htmlTag, ...cssClasses){
```
Comment (gitmnk) [inline portfolio/src/main/webapp/js/script.js:62]: nice use of helper method

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -15,18 +15,47 @@
...
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+
+    List<String> planets = new ArrayList<>();
```
Comment (gitmnk) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Probably  you could do 
List<String> planets = Arrays.asList("Mercury", "Venus", "Earth", "Mars"....);

**Review by gitmnk — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/js/script.js (hunk 5 above)
Comment (qidichen) [reply portfolio/src/main/webapp/js/script.js:37]: Agree, nice comments !

**Review by qidichen — APPROVED** [review]
Good job Benjamin!

**Pull request closed without merging**